src/core/client/framework/lib/network/customErrorMiddleware.ts

```diff
@@ -1,17 +1,32 @@
-import { Middleware } from "react-relay-network-modern/es";
+import { Middleware, RRNLRequestError } from "react-relay-network-modern/es";
 
+import { RelayNetworkRequestError } from "../errors";
+import assertOnline from "./assertOnline";
 import extractGraphQLError from "./extractGraphQLError";
 
+function isRRNLRequestError(error: Error): error is RRNLRequestError {
+  return error.name === "RRNLRequestError";
+}
+
 const customErrorMiddleware: Middleware = (next) => async (req) => {
-  const res = await next(req);
-  if (res.errors) {
-    // Extract custom error.
-    const error = extractGraphQLError(res.errors);
-    if (error) {
-      throw error;
+  try {
+    const res = await next(req);
+    if (res.errors) {
+      // Extract custom error.
+      const error = extractGraphQLError(res.errors);
+      if (error) {
+        throw error;
+      }
+    }
+    return res;
+  } catch (error) {
+    // Make sure we are online, otherwise throw.
+    assertOnline(error);
```

Comment:
Pedantic C-coder nit: An assert is different from throwing an error. Would have named this `throwWhenOffline(...)`. Don't worry about changing it though, not worth the effort.